### PR TITLE
xsum builder

### PR DESCRIPTION
**Pull request by stevengj**
For https://github.com/stevengj/Xsum.jl, replacing https://github.com/stevengj/xsumBuilder

### CODE REVIEW

File: X/xsum/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/xsum
+mkdir -p ${prefix}/lib
+if [[ $target == i686-* ]]; then
+    xsumfpmath="-mfpmath=sse"
+fi
+${CC} ${LDFLAGS} -shared -fPIC -O3 -std=c99 ${xsumfpmath} -fno-inline-functions -o ${prefix}/lib/libxsum.${dlext} xsum.c pbinary.c
```
Comment (giordano) [inline X/xsum/build_tarballs.jl:19]:
```suggestion
${CC} -shared -fPIC -O3 -std=c99 ${xsumfpmath} -fno-inline-functions -o ${libdir}/libxsum.${dlext} xsum.c pbinary.c
```

File: X/xsum/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,36 @@
+using BinaryBuilder
+
+name = "xsum"
+version = v"1.0"
+
+# Collection of sources required to build Xsum
+sources = [
+    "https://gitlab.com/radfordneal/xsum.git" =>
+    "63bd4c13e4a8da4932ded362e2921ed99079f432", # version 2018-12-02 "1.0"
```
Comment (giordano) [inline X/xsum/build_tarballs.jl:9]:
```suggestion
    GitSource("https://gitlab.com/radfordneal/xsum.git",
              "63bd4c13e4a8da4932ded362e2921ed99079f432"), # version 2018-12-02 "1.0"
```

File: X/xsum/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/xsum
+mkdir -p ${prefix}/lib
```
Comment (giordano) [inline X/xsum/build_tarballs.jl:15]:
```suggestion
mkdir -p ${libdir}
```

**Comment by stevengj** [comment]
What's the easiest way to find the UUID of the xsum_jll package?

**Comment by giordano** [comment]
With BinaryBuilder you can do:
```julia
julia> using BinaryBuilder

julia> BinaryBuilder.jll_uuid("xsum_jll")
UUID("e979a739-315a-50ee-b437-b496a9503be1")
```
but arguably the easiest thing to do is to just check out the [`Project.toml` file](https://github.com/JuliaBinaryWrappers/xsum_jll.jl/blob/master/Project.toml)

**Comment by stevengj** [comment]
Thanks, might be good to mention in the README where to find this.

**Pull request merged**